### PR TITLE
Incorporate SymbolToken to SymbolTable

### PR DESCRIPTION
**Pull request by byronlin-bq**
Issue #, if available:

Description of changes:

Add following APIs to SymbolTable.go
```
Find(symbol string)
```

This PR is dependent on #127 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ion/symboltable.go [hunk 1]
```diff
@@ -28,7 +28,8 @@ type SymbolTable interface {
 	Symbols() []string
 	// MaxID returns the maximum ID this symbol table defines.
 	MaxID() uint64
-
+	// Find finds the the SymbolToken by its name.
+	Find(symbol string) (SymbolToken, bool)
```
Comment (fernomac) [inline ion/symboltable.go:32]: Should this return a `*SymbolToken`?

File: ion/symboltable.go [hunk 2]
```diff
@@ -37,6 +38,8 @@ type SymbolTable interface {
 	WriteTo(w Writer) error
 	// String returns an ion text representation of the symbol table.
 	String() string
+	// InternToken adds a new symbol to this symbol table, or finds an existing definition of it.
+	InternToken(symbol string) SymbolToken
```
Comment (fernomac) [inline ion/symboltable.go:42]: `SymbolTable`s are meant to be immutable. Could this go on `SymbolTableBuilder` instead?

Comment (byronlin-bq) [reply ion/symboltable.go]: Removed this API because symboltablebuilder has `Add()`

File: ion/symboltable.go (hunk 1 above)
Comment (byronlin-bq) [reply ion/symboltable.go]: Updated to return *symbolToken.

File: ion/symboltable.go [hunk 3]
```diff
@@ -128,6 +129,20 @@ func (s *sst) Adjust(maxID uint64) SharedSymbolTable {
...
+	if !ok {
+		return nil
+	}
+
+	text, ok := s.FindByID(id)
+	if !ok {
+		return nil
+	}
+
+	return &SymbolToken{Text: &text, LocalSID: SymbolIDUnknown}
```
Comment (justing-bq) [inline ion/symboltable.go:143]: Why are we setting the SID to `SymbolIDUnknown`? What about the `id` that we computed earlier?

File: ion/symboltable.go [hunk 4]
```diff
@@ -318,6 +337,24 @@ func (t *lst) MaxID() uint64 {
...
+		symbolToken := imp.Find(s)
+		if symbolToken != nil {
+			return symbolToken
+		}
+	}
+
+	// Check local
+	_, ok := t.index[s]
+	if ok {
+		return &SymbolToken{Text: &s, LocalSID: SymbolIDUnknown}
```
Comment (justing-bq) [inline ion/symboltable.go:352]: Again why `SymbolIDUnknown`

File: ion/symboltable.go (hunk 3 above)
Comment (byronlin-bq) [reply ion/symboltable.go:143]: According to the Ion specs, it is irrelevant

> SymbolToken APIs, return a SymbolToken with that text and with an undefined importLocation (it is irrelevant).

http://amzn.github.io/ion-docs/guides/symbols-guide.html

Comment (byronlin-bq) [reply ion/symboltable.go:143]: According to Almann: https://github.com/amzn/ion-go/issues/11#issuecomment-580951115

> If the Local SID is defined, the text must be unknown and the source must be undefined. This implies that the symbol token has yet to be resolved and we do not (yet) know the context of such a token.
In all other cases Local SID must be undefined as this implies that the symbol token has been resolved and the local identifier is not meaningful in this context.

File: ion/symboltable.go (hunk 4 above)
Comment (byronlin-bq) [reply ion/symboltable.go:352]: same as above ^

**Comment by byronlin-bq** [comment]
Moved this PR into #145

**Pull request closed without merging**